src/App.js

```diff
@@ -1,16 +1,40 @@
 import React from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog';
+import { useState } from 'react';
 
 const App = () => {
+  const [chatData, setChatData] = useState(chatMessages);
+  const [likes, setLikes] = useState(0);
+
+  const updateLikeStatus = (entryId) => {
+    const updatedEntries = chatData.map(entry => {
+      if (entry.id === entryId) {
+        setLikes(likes + 1);
+        return {
+          ...entry,
+          liked: !entry.liked
+        }
+      };
+      return entry;
+    });
+
+    setChatData(updatedEntries);
+  }
```

Comment:
Great mapping function inside this callback function!

Small bug: `updateLikeStatus` doesn't distinguish between "like" and "unlike" so if user continues to press the message's heart, this likes count will always continue to increment despite the count fluctuating up AND down. How could this code distinguish between a "like" click and a "unlike" click?